City-of-Helsinki/drupal-helfi-etusivu
# UHF-5828 Front page menus

# [UHF-5828](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5828)


## What was done


* Added helfi_navigation module and enabled it
* Added etusivu-specific theme hook suggestions and preprocessing for the external menus.
* Added templates for the etusivu-instance menus.

## How to install

* Make sure your instance is up and running on correct branches.
  * `git pull origin UHF-5828_front_page_menus`
  * `composer require drupal/helfi_api_base:dev-UHF-6197 -W`
    * If the installation fails, you can use `composer reinstall drupal/helfi_api_base --prefer-source`
    * Then go to `./public/modules/contrib/helfi_api_base` and switch to correct branch to `git fetch && git checkout UHF-6197`
  * `composer require drupal/hdbt:dev-UHF-6038-menu-data-syncing-compiled-throw-away`
  * `make fresh`
* Run `make drush-cr`
* To create menu links for the global menu without having to install every instance to your local, do the following:
  * Download this file snapshot of global menu db tables: [global_menu.sql.zip](https://github.com/City-of-Helsinki/drupal-helfi-etusivu/files/9470495/global_menu.sql.zip)
  * Place the contents of the zip to the root of the drupal-helfi-etusivu project
  * Run `make shell`
    * In shell run `drush sql:query --file=/app/global_menu.sql`
     
## How to test


* [ ] Check that the main navigation and mega menu works
   * There should be links to the sub-instances   
![image](https://user-images.githubusercontent.com/1712902/188068536-5b181993-a9f7-4a39-a48b-c2bb1211fed7.png)
* [ ] Check that the mobile fallback navigation works 
   * Open this page in incognito mode: https://helfi-etusivu.docker.so/fi/uutiset
   * Resize the window in to a mobile width. Open inspection tool and hit CMD + SHIFT + P. Type in disable javascript and hit enter
   * Refresh the page and check that the mobile menu looks like this
![image](https://user-images.githubusercontent.com/1712902/188069113-9777b487-fc40-4545-b94f-84b1c6f26763.png)




## Designers review



## Other PRs


* Link to other PR
